tests/read_group/test_read_group_model.py

```diff
@@ -0,0 +1,176 @@
+from dataservice.api.participant.models import Participant
+from dataservice.api.biospecimen.models import Biospecimen
+from dataservice.api.genomic_file.models import GenomicFile
+from dataservice.api.sequencing_center.models import SequencingCenter
+from dataservice.api.read_group.models import ReadGroup
+from dataservice.api.sequencing_experiment.models import SequencingExperiment
+from tests.utils import IndexdTestCase
+from tests.mocks import MockIndexd, MockResp
+
+
+class ModelTest(IndexdTestCase):
+    """
+    Test database model
+    """
+    def create_read_group(self):
+        """
+        create sequencing_center and
+        sequencing experiment save the above entities to db
+        returns sequencing_experiment_id
+        """
+        sc = SequencingCenter(name="Baylor")
+        se_id = "Test_SequencingExperiment_0"
+        seq_experiment_data = self._make_seq_exp(external_id=se_id)
+        se = SequencingExperiment(
+            **seq_experiment_data, sequencing_center_id=sc.kf_id)
+        sc.sequencing_experiments.extend([se])
+        db.session.add(sc)
+        db.session.commit()
+        ids = {'sequencing_experiment_id': se_id}
+        return ids
+
+    def test_create_and_find_read_group(self):
+        """
+        Test creation of read_group 
+        """
+        self._create_entities()
+        gf = GenomicFile.query.first()
+        rg = ReadGroup(genomic_file_id=gf.kf_id)
+        db.session.add(rg)
+        db.session.commit()
+
+        self.assertEqual(GenomicFile.query.count(), 1)
+        self.assertEqual(ReadGroup.query.count(), 1)
+        self.assertEqual(ReadGroup.query.first().genomic_file, gf)
+
+    def test_no_gf(self):
+        """
+        Test creation of read_group without a genomic_file
+        """
+        rg = ReadGroup()
+        self.assertRaises(IntegrityError, db.session.add(rg))
+
+    def test_update_read_group(self):
+        """
+        Test Updating read_group 
+        """
+        self._create_entities()
+        gf = GenomicFile.query.first()
+        rg = ReadGroup(genomic_file_id=gf.kf_id)
+        db.session.add(rg)
+        db.session.commit()
+
+        self.assertEqual(ReadGroup.query.count(), 1)
+        rg = ReadGroup.query.first()
+        rg.paired_end = 1
+        rg.flow_cell = 8
+        db.session.add(rg)
+        db.session.commit()
+
+        rg = ReadGroup.query.first()
+        self.assertEqual(rg.paired_end, 1)
+        self.assertEqual(rg.flow_cell, '8')
+
+    def test_delete_read_group(self):
+        """
+        Test Deleting sequencing_experiment
+        """
+        self._create_entities()
+        gf = GenomicFile.query.first()
+        rg = ReadGroup(genomic_file_id=gf.kf_id)
+        db.session.add(rg)
+        db.session.commit()
+
+        self.assertEqual(ReadGroup.query.count(), 1)
+
+        db.session.delete(rg)
+        db.session.commit()
+
+        self.assertEqual(ReadGroup.query.count(), 0)
+
+    def test_delete_orphans(self):
```

Comment:
Maybe also add a test to ensure the DB cascading delete works. If you do db.session.delete, it will use the ORM's cascading delete functionality. If you do GenomicFile.query.filter_by(kf_id=<gf id>).delete() it will use the DB's delete and then you can check if the cascade worked